src/skmatter/decomposition/_pcovr.py

```diff
@@ -637,6 +637,10 @@ def score(self, X, y, T=None):
         """
         X, y = validate_data(self, X, y, reset=False)
 
+        if y.ndim == 2:
+            if y.shape[1] == 1:
+                y = y.ravel()
```

Comment:
Is this really necessary here? We are changing the user data and also the output dimension. This may lead to shape problems later?

```suggestion
        if y.ndim == 2 and y.shape[1] == 1:
            y = y.ravel()
```

Reply:
I could use your input on this - I just took a look at some sklearn implementations - it seems like `Ridge().fit(X, y).predict(X)` will return a flattened array and have a 1d `coef_` even if trained on 2d y. This is shared by `Lasso()` but not by `LinearRegression()`, which will keep the same shape of input y. This issue shows up in the `PCovR.fit` call when we call`check_lr_fit` to see if the shape of coef_ is compatible with the shape of y. Do you have any thoughts?

Reply:
Hold on, now we have a new problem. Since `LinearRegression()` will have 2d `coef_`, if we fit PCovR with a prefit `LinearRegression()` these lines will cause an error. I'll make a fix for that, but hopefully we can make the code more consistent when each of the sklearn regressors have more similar behavior.

Reply:
uff, yeah we can hotfix for it now. These shapes inconsistency are very nasty...

> more consistent when each of the sklearn regressors have more similar behavior.

You mean on our side or is this something on sklearn side to be changed?

Reply:
On sklearn's side, it looks to me like the behaviors of the sklearn linear regressions aren't consistent with this right now :(

Reply:
Made better fixes now so that PCovR will maintain the same behavior